adapter/oam.go

```diff
@@ -1,8 +1,133 @@
 package adapter
 
-import "context"
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	backoff "github.com/cenkalti/backoff/v4"
+)
 
 // ProcessOAM processes OAM components. This is adapter specific and needs to be implemented by each adapter.
 func (h *Adapter) ProcessOAM(context.Context, OAMRequest) (string, error) {
 	return "", nil
 }
+
+// OAMRegistrant provides utility functions for registering
```

Comment:
The bifurcation of treatment of oam-operations and current operations is concerning: one is registered, while the other is discovered.